src/common/dbus/platform/asio_libdbus/dbus.cpp

```diff
@@ -67,15 +67,14 @@ void HandleReply(DBusPendingCall *pending, void *data);
 DBusHandlerResult MsgFilter(DBusConnection *connection, DBusMessage *message, void *data);
 
 error::Error DBusPeer::InitializeConnection() {
-	DBusError dbus_error;
-	dbus_error_init(&dbus_error);
-	dbus_conn_.reset(dbus_bus_get_private(DBUS_BUS_SYSTEM, &dbus_error));
+	std::unique_ptr<DBusError, void (*)(DBusError *)> dbus_error {new DBusError, dbus_error_free};
```

Comment:
I support the incentive, but I suspect you have in fact have introduced a memory leak here. There are two levels of resources here, the `DBusError` struct itself, and the content it contains. `dbus_error_free` only frees the latter.

I can see two ways to solve it:

1. Use a lambda with both `dbus_error_free` and `delete dbus_error` in it.
2. Don't use `new DBusError`, but keep the variable on the stack instead, as before, and use `&dbus_error` as the argument when constructing the smart pointer (with a different name), together with `dbus_error_free`. The `unique_ptr` now doesn't own the pointer itself, but it owns the contents.

Same for the other commit.

Reply:
Oh shite. True. Rookie mistake!